### PR TITLE
Change unsubscribe UI to email settings

### LINKED ISSUE #369: Change "unsubscribe" UI to "email settings"
In #329 we added the feature to allow users to unsubscribe from course emails, but "unsubscribe" is not very clear to users. Instead, we'd like to use the same UI that edX does in edx-platform:

#### Designs and Mockups

Here's a screenshot of the pop-up on edX:

image

#### Acceptance Criteria:

- [ ] Instead of "unsubscribe" change the menu item to "email settings" 
- [ ] When email settings is selected, show a pop-up with:
   - [ ] a title of "Email settings for <course number>" (or course name if course number isn't an option right now)
   - [ ] a checkbox labelled "Receive course emails" 
   - [ ] a save button
- [ ] When the pop is displayed, the state of the checkbox should reflect the state of the user's subscription, i.e. it should be checked if they are currently subscribed and unchecked if they are not. 
 #### Out of Scope

- We need a way for users to unsubscribe from all emails, including marketing emails

### PR DESCRIPTION
**Pull request by umarmughal824**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
fixes #369 

#### What's this PR do?
Change unsubscrib UI to email settings

#### How should this be manually tested?
(Required)

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
Screenshot 2022-01-19 at 18 42 23

Screenshot 2022-01-19 at 18 42 32
Screenshot 2022-01-19 at 18 42 40
Screenshot 2022-01-19 at 18 42 48
Screenshot 2022-01-19 at 18 42 58

### CODE REVIEW

**Comment by umarmughal824** [comment]
@pdpinch I guess we should also introduce pop for the un-enrollment. I don't see any ticket for it. I am not sure, either it's needed or not.

**Comment by pdpinch** [comment]
If we can't display the course number (code introduced by @MAbdurrehman12 in https://github.com/mitodl/mitxonline/pull/333) then please use the course title. 

Course key / course id is just too confusing for users.

**Comment by umarmughal824** [comment]
@pdpinch that's how it looks like now

Screenshot 2022-01-20 at 15 02 44

File: static/js/containers/pages/DashboardPage.js [hunk 1]
```diff
@@ -3,12 +3,18 @@
...
   Tooltip,
   Dropdown,
   DropdownToggle,
   DropdownMenu,
-  DropdownItem
+  DropdownItem,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
```
Comment (arslanashraf7) [inline static/js/containers/pages/DashboardPage.js:16]: Unused.

File: static/js/containers/pages/DashboardPage.js [hunk 2]
```diff
@@ -173,6 +192,59 @@ export class DashboardPage extends React.Component<
...
+        <ModalHeader toggle={this.toggleEmailSettingsModalVisibility}>
+          Email Settings for {enrollment.run.course_number}
+        </ModalHeader>
+        <ModalBody>
+          <Formik
+            onSubmit={this.onSubmit.bind(this)}
+            initialValues={{
+              subscribeEmails: enrollment.edx_emails_subscription,
+              enrollmentId:    enrollment.id,
+              courseNumber:    enrollment.run.course_number
```
Comment (arslanashraf7) [inline static/js/containers/pages/DashboardPage.js:212]: We should add `course_number` in [courseTypes.js](https://github.com/mitodl/mitxonline/blob/main/static/js/flow/courseTypes.js#L29-L32)

**Review by pdpinch — APPROVED** [review]
screenshots look good to me

**Review by arslanashraf7 — APPROVED** [review]
👍 looks good and works well, Just a couple of comments. feel free to merge afterwards.

File: static/js/factories/course.js [hunk 3]
```diff
@@ -34,7 +34,8 @@ export const makeCourseRun = (): CourseRun => ({
   courseware_id:    casual.word.concat(genCoursewareId.next().value),
   run_tag:          casual.word.concat(genRunTagNumber.next().value),
   // $FlowFixMe
-  id:               genCourseRunId.next().value
+  id:               genCourseRunId.next().value,
+  course_number:    casual.word
```
Comment (umarmughal824) [inline static/js/factories/course.js:38]: @arslanashraf7 see here it's added

**Pull request merged**